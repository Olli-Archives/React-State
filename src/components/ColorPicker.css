.ColorPicker{
  .button{
    width:300px;
  }
  .red{
    background-color:red;
  }
  .green{
    background-color:green;
  }
  .blue{
    background-color:blue;
  }
}