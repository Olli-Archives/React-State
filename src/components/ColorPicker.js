import React, { PureComponent } from 'react';
import styles from './ColorPicker.css';
import PropTypes from 'prop-types';



export default class ColorPicker extends PureComponent{

  static propTypes = {
    color: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
  }

  //{this.props.onClick.bind(null, this.props.color)}
  render(){
    const { color } = this.props;
    return (
      <section className={styles.ColorPicker}>
        <button className={styles[color]} onClick={this.props.onClick.bind(null, color)} >{this.props.color}</button> 
      </section>
    );
  }
}
