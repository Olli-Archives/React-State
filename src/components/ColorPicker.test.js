import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorContainer';

describe('color picker component', ()=>{
  it('renders color picker component', ()=>{
    const wrapper = shallow(<ColorPicker/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
