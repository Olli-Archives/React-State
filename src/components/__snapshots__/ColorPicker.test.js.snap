// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color picker component renders color picker component 1`] = `
<div>
  <ColorPicker
    color="red"
    onClick={[Function]}
  />
  <ColorPicker
    color="blue"
    onClick={[Function]}
  />
  <ColorPicker
    color="green"
    onClick={[Function]}
  />
  <ColorDiv
    clicks={0}
    color=""
  />
</div>
`;
